hack/update-staging-godeps-dockerized.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# updates the godeps.json file in the staging folders to allow clean vendoring
+# based on kubernetes levels.
+# TODO this does not address client-go, since it takes a different approach to vendoring
+# TODO client-go should probably be made consistent
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+# Sets default values
+DRY_RUN=false
+FAIL_ON_DIFF=false
+GODEP_OPTS=""
+
+# Controls verbosity of the script output and logging.
+KUBE_VERBOSE="${KUBE_VERBOSE:-5}"
+if (( ${KUBE_VERBOSE} >= 6 )); then
+  GODEP_OPTS+=("-v")
+fi
+
+while getopts ":df" opt; do
+  case $opt in
+    d) # do not godep-restore into a temporary directory, but use the existing GOPATH
+      DRY_RUN=true
+      ;;
+    f) # fail if something in the Godeps.json files changed
+      FAIL_ON_DIFF=true
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+
+kube::util::ensure_godep_version v79
+
+echo "Checking whether godeps are restored"
+if ! kube::util::godep_restored 2>&1 | sed 's/^/  /'; then
```

Comment:
I'm not actually persisting the storage here, so no, this doesn't cache the godeps atm.